### PR TITLE
Ports: Adds Another World VM interpreter implementation.

### PR DESCRIPTION
**Pull request by safarp**
Making the [Another World Bytecode Interpreter](https://github.com/fabiensanglard/Another-World-Bytecode-Interpreter) to compile and link. Most of the change is to find SDL2 headers and library.

Game runs when game data from the MS-DOS version are provided to the installation directory (see install() in the package.sh).

### CODE REVIEW

**Comment by gunnarbeutner** [comment]
Can you also please add the new port to `Ports/AvailablePorts.md`? That's why the CI run for this currently fails.

**Comment by gunnarbeutner** [comment]
This looks good as far as I'm concerned. Can you squash those two commits though? There shouldn't be any commits for fixing things that were introduced in the same PR. :)

File: Ports/Another-World/package.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+useconfigure=true
+version=git
+depends="SDL2 zlib"
+workdir=Another-World-Bytecode-Interpreter-master
+configopts="-DCMAKE_TOOLCHAIN_FILE=${SERENITY_SOURCE_DIR}/Toolchain/CMake/CMakeToolchain.txt"
+files="https://github.com/fabiensanglard/Another-World-Bytecode-Interpreter/archive/refs/heads/master.zip master.zip 326de7622e5f83a83fce76e6032240157a9dde83c0d65319095c7e0b312af317"
+auth_type=sha256
+launcher_name="Another World"
+launcher_category=Games
+launcher_command="/opt/Another_World/raw --datapath=/opt/Another_World"
```
Comment (linusg) [inline Ports/Another-World/package.sh:12]: Can we please use a dash for the folder name? Underscore is very uncommon across the system.

File: Ports/Another-World/patches/cmake/FindSDL2.cmake [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	# Set the temp variable to INTERNAL so it is not seen in the CMake GUI
+	SET(SDL2_LIBRARY_TEMP "${SDL2_LIBRARY_TEMP}" CACHE INTERNAL "")
+ENDIF(SDL2_LIBRARY_TEMP)
+
+message("</FindSDL2.cmake>")
+
+INCLUDE(FindPackageHandleStandardArgs)
+
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(SDL2 REQUIRED_VARS SDL2_LIBRARY SDL2_INCLUDE_DIR)
+
```
Comment (linusg) [inline Ports/Another-World/patches/cmake/FindSDL2.cmake:169]: Why is this needed? There's a bunch of other ports using CMake and SDL2, and they don't have this. It's also not a patch, so doesn't belong here either :thinking:

**Review by linusg — CHANGES_REQUESTED** [review]

File: Ports/Another-World/package.sh (hunk 1 above)
Comment (safarp) [reply Ports/Another-World/package.sh]: Absolutely. I just pushed the change to the PR.

I took [Super-Mario](https://github.com/SerenityOS/serenity/blob/master/Ports/Super-Mario/package.sh) as an example, it is using underscore in the folder name.

File: Ports/Another-World/patches/cmake/FindSDL2.cmake (hunk 2 above)
Comment (safarp) [reply Ports/Another-World/patches/cmake/FindSDL2.cmake:169]: I will need to investigate this little bit more and will get back to you. Had troubles with linking SDL2 (library wasn't found) so I did the same as [Super-Mario](https://github.com/Bennyhwanggggg/Super-Mario-Clone-Cpp/tree/master/cmake) port for SerenityOS is using.

Comment (safarp) [reply Ports/Another-World/patches/cmake/FindSDL2.cmake]: I have updated the port so that FindSDL2.cmake is no longer needed.
